docs/developer-guide/plugins-scenarios.md

```diff
@@ -233,115 +231,79 @@ However there are some pitfalls you need to know:
+
 ```java
 public class AsyncResponseContentStreamTransformation implements Plugin {
     @Override
-    public StyxObservable<HttpResponse> intercept(HttpRequest request, Chain chain) {
+    public Eventual<LiveHttpResponse> intercept(LiveHttpRequest request, Chain chain) {
         return chain.proceed(request)
-                .map(response -> {
-                        StyxObservable<ByteBuf> contentMapping = response.body()
-                                .map(buf -> {
-                                    buf.release();
-                                    return copiedBuffer(buf.toString(UTF_8).toUpperCase(), UTF_8);
-                                });
-                        return response.newBuilder()
-                                .body(contentMapping)
-                                .build();
-                });
+            .map(response ->                                                                        // 1
+                response.newBuilder()                         
+                    .body(byteStream ->                                                             // 2
+                            byteStream.map(buf -> {                                                 // 3
+                                String upperCase = new String(buf.content(), UTF_8).toUpperCase();  // 4
+                                return new Buffer(upperCase, UTF_8);
+                            }))
+                    .build());
     }
 }
 ```
 
-### Asynchronously transforming streaming response content
+This is quite similar to the request transformation but it involves an additional lambda
+expression to capture the HTTP response from `Eventual` envelope:
 
-```java
-public class AsyncResponseContentStreamTransformation implements Plugin {
-    @Override
-    public StyxObservable<HttpResponse> intercept(HttpRequest request, Chain chain) {
-        return chain.proceed(request)
-                .map(response -> {
-                        StyxObservable<ByteBuf> contentMapping = response.body()
-                                .flatMap(buf -> {
-                                    String content = buf.toString(UTF_8);
-                                    buf.release();
-                                    return sendToRemoteServer(content)
-                                            .map(value -> copiedBuffer(value, UTF_8));
-                                });
-                        return response.newBuilder()
-                                .body(contentMapping)
-                                .build();
-                });
-    }
+  1. `chain.proceed` returns an `Eventual` HTTP response. We apply a `Eventual.map` operator
+     to tap into this response. The `map` operator accepts a lambda expression that handles the
+     response when it eventually is available.
+     
+After that, we will transform the response just like we did in the previous example for
```

Comment:
Looks like there is some whitespace added between "for" and "response" (newline + spaces?).

Reply:
Ok. Removed.